install_sct

```diff
@@ -348,7 +352,7 @@ for arg in "$@"; do
   esac
 done
 
-while getopts ":iydbcvh" opt; do
+while getopts ":iydbcgvh" opt; do
```

Comment:
I believe you need a colon after `g` in the option spec to capture the following argument as `$OPTARG`:

```suggestion
while getopts ":iydbcg:vh" opt; do
```

Reply:
Thank you for pointing this out! I definitely didn't understand `getopts` syntax as well as I thought I did.

From what I can tell, adding a `:` makes the argument mandatory? But, in most cases, we'll be leaning on the default value, rather than expecting the user to specify the value themselves. (The option, in this case, is mainly to provide support for running the install script standalone within CI workflows, so that we can `git clone` PR branches, as we do in the Windows.)

However, `getopts` [doesn't support optional arguments](https://stackoverflow.com/questions/11517139/optional-option-argument-with-getopts). There do seem to be workarounds. though?

- Write a function that provides this functionality: https://stackoverflow.com/a/57295993
- Use `getopt` (no `s`) instead: https://stackoverflow.com/a/52674277
    - Though, there are big cross-platform compatibility concerns here, so I think `getopt` is a no-go:
      > getopt vs getopts .. very different cross platform compliance
    
      > While I too find GNU getopt superior, it's not built in on BSD systems (like macOS) so if your goal is cross platform compatibility, go with getopts

We could also ditch the notion of an option entirely, and instead use an environment variable for this, given that this is only really relevant for our CI in particular? (In fact, I'm a little surprised I didn't do this in the Windows install script...)

Reply:
Scratch this -- We don't need a `-g` option at all. I was mistaken: https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4049#discussion_r1118965144

Reply:
I think "mandatory argument" means something different here: not that `-g` must always be specified, but rather that _its_ argument (the thing after `-g`, in this case the git ref we want to clone) has to be specified if `-g` is present. In our case, it would make no sense to say `-g` without giving a git ref, so it shouldn't matter to us.

Reply:
> I think "mandatory argument" means something different here: not that `-g` must always be specified, but rather that _its_ argument (the thing after `-g`, in this case the git ref we want to clone) has to be specified if `-g` is present.

You're right! My apologies... I had sworn I had come across a `getopts` example where an error was thrown when an argument wasn't passed, but perhaps I was reading some custom bit of bash that someone has written. My bad. :sweat: